cpp/autosar/src/rules/A5-1-1/LiteralValueUsedOutsideTypeInit.ql

```diff
@@ -18,6 +18,25 @@ import cpp
 import codingstandards.cpp.autosar
 import codingstandards.cpp.LoggingOperation
 import codingstandards.cpp.Literals
+import codingstandards.cpp.standardlibrary.FileStreams
+
+/**
+ * In a wrapper `Function`, all accesses of all `Parameters`
+ * are in located in logging or stream calls
+ */
+class LoggerOrStreamWrapperFunction extends Function {
+  LoggerOrStreamWrapperFunction() {
+    forall(Parameter p | p.getFunction() = this |
+      forall(VariableAccess va | va = p.getAnAccess() |
+        (
+          any(FileStreamFunctionCall fc).getAnArgument().getAChild*() = va
+          or
+          any(LoggingOperation logOp).getALoggedExpr().getAChild*() = va
+        )
+      )
+    )
+  }
+}
```

Comment:
another quick pls double check me on this (after looking at the below comment), but the query wouldnt need the  "`LoggingOperationWrapperExpression` from the arguments." I think, since it would be required to implement `getALoggedExpr()`